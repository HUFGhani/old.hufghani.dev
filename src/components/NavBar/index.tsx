import { Box, Flex, Stack, useColorMode } from '@chakra-ui/react'
import ThemeToggle from '@components/themeToggle'
import { Link } from 'gatsby'
import React from 'react'

const NavBar = () => {
  return (
    <Flex as="nav" align="center" justify="flex-end" wrap="wrap" w="100%" p="8" bg="transparent">
      <MenuLinks />
    </Flex>
  )
}

const MenuItem = (props: { [x: string]: any; children: string; to: string }) => {
  const { colorMode } = useColorMode()
  const { children, isLast, to, close, ...rest } = props
  return (
    <Box
      borderRadius="md"
      padding="2.5"
      sx={{
        _hover: {
          backgroundColor:
            colorMode === 'light' ? 'var(--chakra-colors-gray-100)' : 'var(--chakra-colors-whiteAlpha-200)',
        },
      }}
    >
      <Link to={to} {...rest}>
        {children}
      </Link>
    </Box>
  )
}

const MenuLinks = () => {
  return (
    <Stack spacing={8} align="center" justify="flex-end" direction="row" pt="0">
      <MenuItem to="/">Home</MenuItem>
      <MenuItem to="/project">Project</MenuItem>
      <MenuItem to="/about-me">About Me</MenuItem>
      <ThemeToggle />
    </Stack>
  )
}

export default NavBar
