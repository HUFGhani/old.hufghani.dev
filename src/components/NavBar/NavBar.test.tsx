import { ChakraProvider, CSSReset } from '@chakra-ui/react'
import { fireEvent, render } from '@testing-library/react'
import * as Gatsby from 'gatsby'
import React from 'react'
import NavBar from '.'

const renderNavBar = () => {
  return render(
    <ChakraProvider>
      <CSSReset />
      <NavBar />
    </ChakraProvider>
  )
}
const useStaticQuery = jest.spyOn(Gatsby, 'useStaticQuery')
useStaticQuery.mockImplementation(() => ({
  site: {
    siteMetadata: {
      title: 'Hamza U. F. Ghani',
      description: 'A detailed account of what Hamza U. F. Ghani is about.',
      siteUrl: 'https://hufghani.dev',
      twitterUsername: '@the_ghani',
      authorName: 'Hamza U. F. Ghani',
      siteLanguage: 'en-GB',
      siteLocale: 'en_gb',
      navLinks: {
        menu: [
          {
            name: 'Project',
            url: '/project',
          },
          {
            name: 'About Me',
            url: '/about',
          },
        ],
      },
    },
  },
  allFile: {
    edges: [
      {
        node: {
          publicURL: '/static/51db75c9f2982f1838540680b66abc57/CV.pdf',
        },
      },
    ],
  },
}))

beforeEach(() => {
  jest.clearAllMocks()
})

describe(`<Header/> with feature flag on`, () => {
  it(`should render Header`, () => {
    const { asFragment } = renderNavBar()
    expect(asFragment()).toMatchSnapshot()
  })

  it(`should render correctly and switch the theme from light to dark`, () => {
    const { getByLabelText } = renderNavBar()
    expect(document.querySelector(`body`)).toHaveClass(`chakra-ui-light`)
    fireEvent.click(getByLabelText(`Toggle Theme`))
    expect(document.querySelector(`body`)).toHaveClass(`chakra-ui-dark`)
  })
})
