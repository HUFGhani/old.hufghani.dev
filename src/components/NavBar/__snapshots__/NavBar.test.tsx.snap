// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header/> with feature flag on should render Header 1`] = `
<DocumentFragment>
  <nav
    class="css-xajjvo"
  >
    <div
      class="logo css-0"
    >
      <a
        href="/"
      >
        <h1
          class="chakra-heading css-16v0qoy"
        >
          Hamza U. F. Ghani
        </h1>
      </a>
    </div>
    <div
      class="chakra-stack css-1rqb8o"
    >
      <div
        class="css-1w8wasz"
      >
        <a
          href="/"
        >
          Home
        </a>
      </div>
      <div
        class="css-1w8wasz"
      >
        <a
          href="/about"
        >
          About Me
        </a>
      </div>
      <div
        class="css-1w8wasz"
      >
        <a
          href="/project"
        >
          Project
        </a>
      </div>
      <div
        class="css-1hqy0az"
      >
        <a
          class="chakra-link css-10qsrqw"
          href="/static/51db75c9f2982f1838540680b66abc57/CV.pdf"
          rel="noopener noreferrer"
          target="_blank"
        >
          <abbr
            class="chakra-text css-0"
            title="Curriculum Vitae"
          >
            CV
          </abbr>
        </a>
      </div>
      <button
        aria-label="Toggle Theme"
        class="chakra-button css-150nmn8"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="chakra-icon css-1q8md0w"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <g
            fill="none"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          >
            <circle
              cx="12"
              cy="12"
              r="5"
            />
            <path
              d="M12 1v2"
            />
            <path
              d="M12 21v2"
            />
            <path
              d="M4.22 4.22l1.42 1.42"
            />
            <path
              d="M18.36 18.36l1.42 1.42"
            />
            <path
              d="M1 12h2"
            />
            <path
              d="M21 12h2"
            />
            <path
              d="M4.22 19.78l1.42-1.42"
            />
            <path
              d="M18.36 5.64l1.42-1.42"
            />
          </g>
        </svg>
      </button>
    </div>
  </nav>
</DocumentFragment>
`;
